qiskit/backends/local/statevector_simulator_py.py

```diff
@@ -59,9 +60,10 @@ def _run_job(self, qobj):
         self._validate(qobj)
         final_state_key = 32767  # Internal key for final state snapshot
         # Add final snapshots to circuits
-        for circuit in qobj['circuits']:
-            circuit['compiled_circuit']['operations'].append(
-                {'name': 'snapshot', 'params': [final_state_key]})
+        for circuit in qobj.circuits:
+            circuit.compiled_circuit.operations.append(
+                QObjItem.from_dict({'name': 'snapshot', 'params': [final_state_key]})
```

Comment:
I think you should create a `QobjInstruction` here rather than `QobjItem`?

Reply:
Nice catch! Actually needed to tweak `QobjInstruction` a bit, as the snapshot is the only instruction without `qubits`.